### ISSUE TITLE
`useDebounce` can't handle reference type values

### ISSUE THREAD

**Issue by nico-martinucci** [juliencrn/usehooks-ts#392]
Relatively self-explanatory - `useDebounce` drops the value directly into `useEffect`'s dependency array, which uses shallow equality to determine if anything has changed. Fix could be as simple as stringifying the value directly in the dep array:

```
import { useEffect, useState } from 'react';

function useDebounce(value, delay) {
    const [debouncedValue, setDebouncedValue] = useState(value);
    useEffect(() => {
        const timer = setTimeout(() => setDebouncedValue(value), delay || 500);
        return () => {
            clearTimeout(timer);
        };
    }, [JSON.stringify(value), delay]);
    return debouncedValue;
}

export default useDebounce;
```

Happy to make this change, or an alternative solution if one is identified.

**Comment by juliencrn** [comment]
Hi @nico-martinucci, thanks for the feedback. 
We just shipped this #443 that should solve the issue.
It will be released this week

**Comment by nico-martinucci** [comment]
Awesome, thanks @juliencrn !
